src/components/multilineBox/MultilineBox.tsx

```diff
@@ -0,0 +1,132 @@
+    onMount: (defaultIds: string[]) => void;
+    onUnmount: () => void;
+    removeBox: (id: string) => void;
+    addNewBox: () => void;
+}
+
+export interface IMultilineBoxState<T> {
+    initialData: {[id: string]: T};
+}
+
+export interface IMultilineBoxProps<T = any> extends IMultilineBoxOwnProps<T>,
+    Partial<IMultilineBoxStateProps>,
+    Partial<IMultilineBoxDispatchProps> {}
+
+const makeMapStateToProps = () => {
+    const getStateProps = createStructuredSelector({
+        multilineBoxIds: MultilineBoxSelectors.getIds,
+    });
+
+    return (state: IReactVaporState, ownProps: IMultiSelectOwnProps): IMultilineBoxStateProps =>
+        getStateProps(state, {id: ownProps.id});
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IMultilineBoxOwnProps): IMultilineBoxDispatchProps => ({
+    onMount: (defaultIDs: string[]) => dispatch(addStringList(ownProps.id, defaultIDs)),
+    onUnmount: () => dispatch(removeStringList(ownProps.id)),
+    removeBox: (id: string) => dispatch(removeValueStringList(ownProps.id, id)),
+    addNewBox: () => dispatch(addValueStringList(ownProps.id, UUID.generate())),
+});
+
+@ReduxConnect(makeMapStateToProps, mapDispatchToProps)
+export class MultilineBox<T> extends React.PureComponent<IMultilineBoxProps<T>, IMultilineBoxState<T>> {
+
+    constructor(props: IMultilineBoxProps<T>, state: IMultilineBoxState<T>) {
+        super(props, state);
+
+        this.state = {
+            initialData: this.getInitialDataMappedWithBoxIDs(),
+        };
+    }
+
+    private getInitialDataMappedWithBoxIDs(): {[id: string]: T} {
+        const initialData: {[id: string]: T} = {};
+        _.each(this.props.data, (data: T) => {
+            initialData[UUID.generate()] = data;
+        });
+        return initialData;
+    }
+
+    private getInitialBoxesWithAnExtraBox() {
+        const ids: string[] = _.keys(this.state.initialData);
+        ids.push(...this.props.multilineBoxIds, UUID.generate());
+        return ids;
+    }
+
+    private getLastBoxProps(): T {
+        return deepClone(this.props.defaultProps);
+    }
+
+    private getData(): Array<IMultilineSingleBoxProps<T>> {
+        return _.map(this.props.multilineBoxIds, (id: string, index: number) => {
+            const props: T = this.state.initialData[id] || this.getLastBoxProps();
+            return {
+                id,
+                isLast: index === this.props.multilineBoxIds.length - 1,
+                props,
+            };
+        });
+    }
+
+    private getParentProps(): IMultilineParentProps {
+        return {
+            removeBox: (id: string) => this.props.removeBox(id),
+            addNewBox: () => this.props.addNewBox(),
+            parentId: this.props.id,
+        };
+    }
+
+    componentDidUpdate(prevProps: Readonly<IMultilineBoxProps<T>>, prevState: Readonly<IMultilineBoxState<T>>) {
+        if (prevState && !_.isEqual(this.state.initialData, prevState.initialData)) {

```

Comment:
You should check for props change, not state change. See my previous comment for a suggestion.